app/src/main/kotlin/at/bitfire/davdroid/ui/account/AccountScreenModel.kt

```diff
@@ -62,26 +61,32 @@ class AccountScreenModel @AssistedInject constructor(
         fun create(account: Account): AccountScreenModel
     }
 
+    private val _showOnlyPersonal = MutableStateFlow(false)
+    val showOnlyPersonal = _showOnlyPersonal.asStateFlow()
+
+    private var _showOnlyPersonalLocked = MutableStateFlow(false)
+    val showOnlyPersonalLocked = _showOnlyPersonalLocked.asStateFlow()
+
+    /**
+     * Only acquire account settings on a worker thread!
+     */
+    private val accountSettings by lazy { accountSettingsFactory.create(account) }
+
+    private suspend fun reload() = withContext(Dispatchers.Default) {
+        _showOnlyPersonal.value = accountSettings.getShowOnlyPersonal()
+        _showOnlyPersonalLocked.value = accountSettings.getShowOnlyPersonalLocked()
+    }
+
     /** whether the account is invalid and the screen shall be closed */
     val invalidAccount = accountRepository.getAllFlow().map { accounts ->
         !accounts.contains(account)
     }
 
-    private val refreshSettingsSignal = MutableLiveData(Unit)
-    val showOnlyPersonal = refreshSettingsSignal.switchMap<Unit, AccountSettings.ShowOnlyPersonal> {
-        object : LiveData<AccountSettings.ShowOnlyPersonal>() {
-            init {
-                viewModelScope.launch(Dispatchers.IO) {
-                    val settings = accountSettingsFactory.create(account)
-                    postValue(settings.getShowOnlyPersonal())
-                }
-            }
+    fun setShowOnlyPersonal(showOnlyPersonal: Boolean) {
+        CoroutineScope(Dispatchers.Default).launch {
```

Comment:
We have now used `CoroutineScope(Dispatchers.Default)` a few times. My original intention of not using `viewModelScope` was that the write operation can't be interrupted.

However I don't think anymore that this is the model's task. It should just call `repo.setWhatever()` and the called class should make sure that the operation isn't cancellable if this is required.

So it's probably safe to use viewModelScope for all these calls (not only this line).